tests/performance/metrics/metrics.go

```diff
@@ -0,0 +1,162 @@
+type Metric struct {
+	Perc50 float64 `yaml:"perc50"`
+	Perc90 float64 `yaml:"perc90"`
+	Perc99 float64 `yaml:"perc99"`
+	Unit   Unit    `yaml:"unit,omitempty"`
+}
+
+// VMMetricData encapsulates vm startup latency information.
+type VMMetricData struct {
+	// Name of the vm
+	Name string
+	// Node this vm was running on
+	Node string
+	// Metric value
+	Value float64
+	// Metric unit
+	Unit Unit
+	// Type can be latency (e.g., for each VM creation phases), or resource (e.g., of resource usage)
+	Type kvv1.VirtualMachineInstancePhase
+}
+
+// Slice is an array of VMMetricData which encapsulates vm startup latency information.
+type Slice []VMMetricData
+
+func (a Slice) Len() int           { return len(a) }
+func (a Slice) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a Slice) Less(i, j int) bool { return a[i].Value < a[j].Value }
+
+func (a Slice) Sort() Slice { sort.Sort(a); return a }
+
+// ExtractMetrics returns latency metrics for each percentile(50th, 90th and 99th).
+func ExtractMetrics(data []VMMetricData) Metric {
+	metrics := Metric{}
+
+	length := len(data)
+	if length < 1 {
+		return metrics
+	}
+
+	metrics.Perc50 = data[int(math.Ceil(float64(length*50)/100))-1].Value
+	metrics.Perc90 = data[int(math.Ceil(float64(length*90)/100))-1].Value
+	metrics.Perc99 = data[int(math.Ceil(float64(length*99)/100))-1].Value
+	metrics.Unit = data[0].Unit
+	return metrics
+}
+
+// BatchStatus struct contains a map with the VMI metrics and another method for quick lookup
+type BatchStatus struct {
+	sync.RWMutex
+	metrics map[kvv1.VirtualMachineInstancePhase]Slice
+	vmiMap  map[kvv1.VirtualMachineInstancePhase]map[string]kvv1.VirtualMachineInstance
+}
+
+// NewBatchStatus creates a new BatchStatus
+func NewBatchStatus() *BatchStatus {
+	s := &BatchStatus{
+		metrics: make(map[kvv1.VirtualMachineInstancePhase]Slice),
+		vmiMap:  make(map[kvv1.VirtualMachineInstancePhase]map[string]kvv1.VirtualMachineInstance),
+	}
+	s.vmiMap[kvv1.Scheduling] = make(map[string]kvv1.VirtualMachineInstance)
+	s.vmiMap[kvv1.Scheduled] = make(map[string]kvv1.VirtualMachineInstance)
+	s.vmiMap[kvv1.Running] = make(map[string]kvv1.VirtualMachineInstance)
+	s.vmiMap[kvv1.Succeeded] = make(map[string]kvv1.VirtualMachineInstance)
+	return s
+}
+
+// GetVMIPhaseMap returns the list of VMI for a given phase
+func (s *BatchStatus) GetVMIPhaseMap(phase kvv1.VirtualMachineInstancePhase) map[string]kvv1.VirtualMachineInstance {
+	return s.vmiMap[phase]
+}
+
+// GetVMIPhaseMap returns the list of all VMI metrics
+func (s *BatchStatus) GetVMIMetrics() map[kvv1.VirtualMachineInstancePhase]Slice {
+	return s.metrics
+}
+
+// AddVMIPhaseMetrics add metrics for all VMI phases execpt when deleted, which is computed differently
+func (s *BatchStatus) AddVMIPhaseMetrics(vmi kvv1.VirtualMachineInstance) {
+	s.Lock()
+	defer s.Unlock()
+
+	if _, ok := s.vmiMap[kvv1.Running][vmi.ObjectMeta.Name]; !ok {
+
```

Comment:
Sorry for being pedantic. But I don't fully understand why you sometimes have an empty line after the `if` statement and sometimes don't. Is there something I'm missing?

Reply:
Actually, thanks for pointing it. There is not reason to have those line breaks!